### PR TITLE
snapshot support

### PR DESCRIPTION
**Pull request by yasker**
Addressed previous comments. Also test is added.

### CODE REVIEW

File: cattle/__init__.py [hunk 1]
```diff
@@ -298,3 +298,55 @@ def eventlet_backdoor():
...
+        return default_value('VOLMGR_DEFAULT_VOLUME_SIZE', 1073741824)
+
+    @staticmethod
+    def volmgr_mount_dir():
+        return default_value('VOLMGR_MOUNT_DIR', '/var/lib/rancher/mounts')
+
+    @staticmethod
+    def volmgr_dm_data_file():
+        return default_value('VOLMGR_DM_DATA_FILE',
+                             '/var/lib/rancher/volmgr_data_file')
```
Comment (ibuildthecloud) [inline cattle/__init__.py]: put in /var/lib/rancher/volmgr subdir

File: cattle/__init__.py [hunk 2]
```diff
@@ -298,3 +298,55 @@ def eventlet_backdoor():
...
+
+    @staticmethod
+    def volmgr_dm_metadata_file():
+        return default_value('VOLMGR_DM_METADATA_FILE',
```
Comment (ibuildthecloud) [inline cattle/__init__.py]: put in /var/lib/rancher/volmgr subdir

File: cattle/__init__.py [hunk 3]
```diff
@@ -298,3 +298,55 @@ def eventlet_backdoor():
...
+        return default_value('VOLMGR_DM_DATA_DEVICE', '')
+
+    @staticmethod
+    def volmgr_dm_metadata_device():
+        return default_value('VOLMGR_DM_METADATA_DEVICE', '')
+
+    @staticmethod
+    def volmgr_blockstore_dir():
+        return default_value('VOLMGR_BLOCKSTORE_DIR',
+                             '/var/lib/rancher/volmgr_blockstore/')
```
Comment (ibuildthecloud) [inline cattle/__init__.py]: put in /var/lib/rancher/volmgr subdir

**Comment by ibuildthecloud** [comment]
Create issue for idempotency of create and backup

**Comment by ibuildthecloud** [comment]
Can you handle this scenario.

Create snapshot on host A
Delete host A from cattle
Delete snapshot

**Comment by ibuildthecloud** [comment]
Only keep X number of snapshot in DM pool.

**Comment by ibuildthecloud** [comment]
Make initialize off by default

**Comment by yasker** [comment]
Create https://github.com/rancherio/rancher/issues/661 and https://github.com/rancherio/rancher/issues/660 to track two issues.

Since there is concept of blockstore in cattle now, nothing we can do about snapshot after host is gone.  We would do that after we able to add s3 as storage pool for snapshot.

**Pull request merged**